### ISSUE TITLE
Router Firmware Not Pulling IP (Suggested Addition to Script)

### ISSUE THREAD

**Issue by EddieTx** [alphabt/asuswrt-merlin-ddns-cloudflare#6]
My router stopped updating my IP to Cloudflare a while back, but I never looked at the issue until recent after having to try to connect to my house externally.  After a lot of troubleshooting, I finally figured out that the ${1} was not returning my IP, so the router was not pulling it for some reason.  After looking over the original documentation at Merlin's github https://github.com/RMerl/asuswrt-merlin.ng/wiki/Custom-DDNS I figured out that it could be caused by double NAT, so I updated the script and viola.  Below is what I updated, hope it helps.

Added this variable to the top
```
DNS_QUERY_SITE="https://myip.dnsomatic.com/"    # Site used to get external DNS if router isn't able to pull it due to double NAT
```

Added the IF block to test if IPv4 is null, and if so then query external site to obtain IP, also updated the logger to in the failure to include IP as that's how I figured the issue out.
```
# Update IPv4
#Get IPv4 from Router otherwise from external source
IPv4=${1}
if [ -z "$IPv4" ]; then
  IPv4="$(curl -fs4 $DNS_QUERY_SITE)"
  logger "IP ${IPv4} obtained external source $DNS_QUERY_SITE"
else
  logger "IP ${IPv4} obtained by router"
fi

A_RECORD_IDS=$(get_dns_record_ids $RECORD_NAME A $API_TOKEN $ZONE_ID)

for A_RECORD_ID in $A_RECORD_IDS; do
  RESPONSE="$(update_dns_record $RECORD_NAME $A_RECORD_ID A $IPv4 $RECORD_TTL $API_TOKEN $ZONE_ID)"
  echo $RESPONSE | grep '"success":\ *true' >/dev/null

  if [ $? -eq 0 ]; then
    logger "Updated A record for ${RECORD_NAME} to ${IPv4}"
  else
    logger "Unable to update A record for ${RECORD_NAME} with ${IPv4}"
    RESULT=false
  fi
done
```

**Comment by EddieTx** [comment]
I created a fork/patch if you want to commit it.

**Comment by alphabt** [comment]
Thank you for enhancing this Eddit. I am currently a bit busy at work but I'll try my best to review and merge this in sooner than later.

**Comment by alphabt** [comment]
Feel free to create a pull request so we can merge this in.

**Comment by EddieTx** [comment]
Pull request created.

**Comment by alphabt** [comment]
Resolved with #7
